drawee/src/main/java/com/facebook/drawee/drawable/RoundedBitmapDrawable.java

```diff
@@ -217,9 +229,27 @@ private void updateTransform() {
   }
 
   private void updatePath() {
+    mBorderPath.reset();
+    mRootBounds.inset(mBorderWidth / 2, mBorderWidth / 2);
+
+    if (mIsCircle) {
+      mBorderPath.addCircle(
+              mRootBounds.centerX(),
+              mRootBounds.centerY(),
+              Math.min(mRootBounds.width(), mRootBounds.height())/2,
+              Path.Direction.CW);
+    } else {
+      mBorderRadii[0] = mBorderRadii[1] = mCornerRadii[0] + mPadding - mBorderWidth/2;
+      mBorderRadii[2] = mBorderRadii[3] = mCornerRadii[2] + mPadding - mBorderWidth/2;
+      mBorderRadii[4] = mBorderRadii[5] = mCornerRadii[4] + mPadding - mBorderWidth/2;
+      mBorderRadii[6] = mBorderRadii[7] = mCornerRadii[6] + mPadding - mBorderWidth/2;
```

Comment:
We should not assume that x=y for a particular corner.
I am confused that we are applying padding both to the border and to the image. What am I missing?